<launch>
  <arg name="common_param_path" default="$(find-pkg-share obstacle_stop_planner)/config/common.param.yaml"/>
  <arg name="adaptive_cruise_control_param_path" default="$(find-pkg-share obstacle_stop_planner)/config/adaptive_cruise_control.param.yaml"/>
  <arg name="obstacle_stop_planner_param_path" default="$(find-pkg-share obstacle_stop_planner)/config/obstacle_stop_planner.param.yaml"/>

  <!-- vehicle info -->
  <arg name="vehicle_info_param_file" default="$(find-pkg-share autoware_vehicle_info_utils)/config/vehicle_info.param.yaml"/>

  <arg name="input_objects" default="/perception/object_recognition/objects"/>
  <arg name="input_pointcloud" default="input/pointcloud"/>
  <arg name="input_trajectory" default="input/trajectory"/>
  <arg name="input_odometry" default="/localization/kinematic_state"/>
  <arg name="output_trajectory" default="output/trajectory"/>
  <arg name="output_velocity_limit" default="/planning/scenario_planning/max_velocity_candidates"/>
  <arg name="output_velocity_limit_clear_command" default="/planning/scenario_planning/clear_velocity_limit"/>

  <node pkg="obstacle_stop_planner" exec="obstacle_stop_planner_node" name="obstacle_stop_planner" output="screen">
    <!-- load config files -->
    <param from="$(var common_param_path)"/>
    <param from="$(var adaptive_cruise_control_param_path)"/>
    <param from="$(var obstacle_stop_planner_param_path)"/>
    <param from="$(var vehicle_info_param_file)"/>
    <!-- remap topic name -->
    <remap from="~/output/stop_reason" to="/planning/scenario_planning/status/stop_reason"/>
    <remap from="~/output/stop_reasons" to="/planning/scenario_planning/status/stop_reasons"/>
    <remap from="~/output/trajectory" to="$(var output_trajectory)"/>
    <remap from="~/output/max_velocity" to="$(var output_velocity_limit)"/>
    <remap from="~/output/velocity_limit_clear_command" to="$(var output_velocity_limit_clear_command)"/>
    <remap from="~/input/objects" to="$(var input_objects)"/>
    <remap from="~/input/pointcloud" to="$(var input_pointcloud)"/>
    <remap from="~/input/trajectory" to="$(var input_trajectory)"/>
    <remap from="~/input/odometry" to="$(var input_odometry)"/>
  </node>
</launch>
